crates/iota-light-client/src/main.rs

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) Mysten Labs, Inc.
+// Modifications Copyright (c) 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{path::PathBuf, str::FromStr};
+
+use anyhow::Context;
+use clap::{Parser, Subcommand};
+use iota_light_client::{
+    checkpoint::sync_and_verify_checkpoints,
+    config::Config,
+    package_store::RemotePackageStore,
+    verifier::{get_verified_effects_and_events, get_verified_object},
+};
+use iota_package_resolver::Resolver;
+use iota_types::{
+    base_types::ObjectID,
+    digests::TransactionDigest,
+    object::{Data, bounded_visitor::BoundedVisitor},
+};
+use tracing::debug;
+
+// Define the `GIT_REVISION` and `VERSION` consts
+bin_version::bin_version!();
+
+#[derive(Parser, Debug)]
+#[command(
+    name = env!("CARGO_BIN_NAME"),
+    about = env!("CARGO_PKG_DESCRIPTION"),
+    author,
+    version = VERSION,
+    propagate_version = true,
+)]
+struct Args {
+    /// Sets a custom config file
+    #[arg(short, long, value_name = "PATH")]
+    config: Option<PathBuf>,
```

Comment:
Why is it even an option if we panic if it's missing?

Reply:
good point, I think we could keep it optional though and default to `mainnet` now. Wdyt?

Reply:
Yeah why, don't forget to mention it in the doc comment

Reply:
https://github.com/iotaledger/iota/pull/6135/commits/e465a9c1ed5c817e9dd437572fdfd5ee2c04cb00